apps/api.planx.uk/modules/ai/projectDescription/service.ts

```diff
@@ -1,21 +1,98 @@
-import type { ErrorStatus } from "./types.js";
+import {
+  createGateway,
+  generateObject,
+  InvalidPromptError,
+  NoContentGeneratedError,
+  NoObjectGeneratedError,
+  NoSuchModelError,
+} from "ai";
+import { z } from "zod";
 
-type Success = {
-  ok: true;
-  value: string;
-};
-
-type Failure = {
-  ok: false;
-  error: ErrorStatus;
-};
+import {
+  type GatewayResult,
+  GatewayStatus,
+  SUCCESS_STATUSES,
+} from "./types.js";
 
-type Result = Success | Failure;
+const DEFAULT_MODEL = "google/gemini-2.5-pro";
+const SYSTEM_PROMPT_ENHANCE = `Act as a trained planning officer at a local council in the UK.
+You will be prompted with a description of a planning application which is to be submitted to a local council.
+Please improve it such that it has the best chance of being accepted and validated, without adding unneccessary detail.
+If the description does not seem to be related to a planning application, respond with the status $${GatewayStatus.INVALID}.
+If the description is already good enough, return the original without amendment, with the status $${GatewayStatus.NO_CHANGE}.
+If the description can be improved, return your amended version, with the status $${GatewayStatus.ENHANCED}.`;
```

Comment:
I'd be very interested in exposing this as an `.md` file.

This would mean a few things - 
 - Easier iteration for non-GH OSL users
 - We can more easily reference and link to it
 - Separation of concerns - splitting out business logic and the prompt more explicitly
 - More structure, more easily - headings, bullet points, etc. This can all get done now, but a little less easily
 - Sets up a simpler pattern if we want a future shared location (e.g. a `/packages/prompts` within the monorepo)

In all fairness a few of these feel slightly YAGNI, but it seems better to set up a good pattern now.